osquery/config/config.cpp

```diff
@@ -61,278 +106,85 @@ Status Config::update(const std::map<std::string, std::string>& config) {
     }
   }
 
-  // Request a unique write lock when updating config.
-  {
-    boost::unique_lock<boost::shared_mutex> unique_lock(getInstance().mutex_);
-
-    for (const auto& source : config) {
-      if (Registry::external()) {
-        VLOG(1) << "Updating extension config with source: " << source.first;
-      } else {
-        VLOG(1) << "Updating config with source: " << source.first;
-      }
-      getInstance().raw_[source.first] = source.second;
-    }
-
-    // Now merge all sources together.
-    ConfigData conf;
-    for (const auto& source : getInstance().raw_) {
-      auto status = mergeConfig(source.second, conf);
-      if (getInstance().force_merge_success_ && !status.ok()) {
-        return Status(1, status.what());
-      }
-    }
-
-    // Call each parser with the optionally-empty, requested, top level keys.
-    getInstance().data_ = std::move(conf);
-  }
-
-  for (const auto& plugin : Registry::all("config_parser")) {
-    auto parser = std::static_pointer_cast<ConfigParserPlugin>(plugin.second);
-    if (parser == nullptr || parser.get() == nullptr) {
+  for (const auto& source : config) {
+    // load the config (source.second) into a ptree
+    std::stringstream json;
+    json << source.second;
+    ptree tree;
```

Comment:
Do you think following the existing conventions and using `pt::ptree` would make more sense? It may provide more clarity, using this code block as an example the `read_json` and caught exception would be observably related. :dancer: